.github/workflows/build-test-radosgw.yml

```diff
@@ -113,3 +133,98 @@ jobs:
           ./sfs-smoke-test.sh 127.0.0.1:7480
 
           docker kill $CONTAINER
+
+  run-s3gw-tests:
+    strategy:
+      matrix:
+        op:
+          - create
+          # - list
+          - delete
+        resource:
+          - bucket
+          - object
+
+    runs-on: self-hosted
+    needs:
+      - build-and-test
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: ceph
+
+      - name: Checkout s3tests
+        uses: actions/checkout@v3
+        with:
+          repository: ceph/s3-tests
+          path: s3tests
+          branch: ceph-master
+
+      - name: Checkout s3gw Infrastructure
+        uses: actions/checkout@v3
+        with:
+          path: s3gw
+          repository: aquarist-labs/s3gw
+
+      - name: Download Radosgw Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ needs.build-and-test.outputs.artifact_id }}
+
+      - name: Unpack Artitfacts
+        run: |
+          tar -xvf build-results.tar.gz
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+
+      - name: Install dependencies
+        run: |
+          YQ_GH_URL=https://github.com/mikefarah/yq
+          sudo apt-get update
+          sudo apt-get install -y \
+            python3-nose \
+            wget
+
+          # Unfortunately, since yq is only available through snap on Ubunut and
+          # that doesn't work in docker containers (at least not out of the
+          # box), this abomination is the way to go to install yq.
+          sudo wget \
+            "${YQ_GH_URL}/releases/latest/download/yq_linux_amd64" \
```

Comment:
This is basically code execution in our build workers by the yq project.
latest may point to anything. For basic security we may want to pin to version and do a checksum check. Better: replace with python (see below)

Reply:
Ack. See above